src/Spec-MorphicAdapters/MorphicButtonAdapter.class.st

```diff
@@ -56,6 +56,17 @@ MorphicButtonAdapter >> buildWidget [
 		dragEnabled: self dragEnabled;
 		dropEnabled: self dropEnabled ;	
 		eventHandler: (MorphicEventHandler new on: #keyStroke send: #keyStroke:fromMorph: to: self).
+	
+	normalColorBlock := [ :aButton |
+							(aButton valueOfProperty: #noFill ifAbsent: [false]) 
+								ifTrue: [ SolidFillStyle color: Color transparent ]
+								ifFalse: [ SolidFillStyle color: self color ] ].
+	clickedColorBlock := [ :aButton | SolidFillStyle color: self color muchDarker ].
+	button theme: ((UIThemeDecorator theme: button theme)
+							property: #buttonNormalFillStyleFor: returnsValueOf: normalColorBlock;
+							property: #buttonMouseOverFillStyleFor: returnsValueOf: normalColorBlock;
+							property: #buttonPressedFillStyleFor: returnsValueOf: clickedColorBlock;
+							yourself).
```

Comment:
I don't quite like the usage of blocks here...

Reply:
Look at the implementation of `UITheme>>#buttonNormalFillStyleFor:`, I just wanted to keep a similar behaviour.

However, if it is ok to do it, `UIThemeDecorator` allows one to do something like:
```
themeDecorator property: #buttonNormalFillStyleFor: returnsValueOf: (SolidFillStyle color: Color red)
```